lib/screens/authentication/sign_up/viewmodels/signup_bloc.dart

```diff
@@ -174,6 +175,8 @@ class SignupBloc extends Bloc<SignupEvent, SignupState> {
 
     if (!result.isError) {
       _authenticationBloc.add(OnCreateAccount(account: username, privateKey: privateKey.toString()));
+      /// In case there was a recovery in place. We cancel it.
+      StopRecoveryUseCase().run();
     }
```

Comment:
This looks perfect but the cancel recovery may need to happen before we add the OnCreateAccount account event.

Reply:
Checked - this needs to happen before the OnCreateAccount is fired so just move it up 2 lines

Reason: OnCreateAccount initiates the chain of events that causes the auth state to be re-evaluated

When the auth state is re-evaluated, the recovery flag is checked.

Reply:
Nice catch